### PR TITLE
Implement Pl3xMap Hook.

### PR DESCRIPTION
**Pull request by AAigars**
This pull request implements a Pl3xMap hook that allows VanishNoPacket to hide the player on the map whenever they vanish.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <packaging>jar</packaging>
 
     <name>VanishNoPacket</name>
-    <version>3.22-SNAPSHOT</version>
+    <version>3.23-SNAPSHOT</version>
```
Comment (mbax) [inline pom.xml:9]: Why bump when 3.22 hasn't come out yet?

File: src/main/java/org/kitteh/vanish/VanishPlugin.java [hunk 2]
```diff
@@ -259,6 +259,9 @@ public void onEnable() {
             // Shouldn't happen here, but if the load order gets broken...
             this.hookManager.getHook(HookType.DiscordSRV).onEnable();
         }
+        if(this.getConfig().getBoolean("hooks.pl3xmap", false)) {
```
Comment (mbax) [inline src/main/java/org/kitteh/vanish/VanishPlugin.java:262]: Missing space after `if`

File: src/main/java/org/kitteh/vanish/hooks/HookManager.java [hunk 3]
```diff
@@ -22,10 +22,7 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 import org.kitteh.vanish.Debuggle;
 import org.kitteh.vanish.VanishPlugin;
-import org.kitteh.vanish.hooks.plugins.DiscordSRVHook;
-import org.kitteh.vanish.hooks.plugins.DynmapHook;
-import org.kitteh.vanish.hooks.plugins.EssentialsHook;
-import org.kitteh.vanish.hooks.plugins.VaultHook;
+import org.kitteh.vanish.hooks.plugins.*;
```
Comment (mbax) [inline src/main/java/org/kitteh/vanish/hooks/HookManager.java:25]: No thanks.

File: src/main/java/org/kitteh/vanish/hooks/plugins/Pl3xMapHook.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        } else {
+            this.plugin.getLogger().info("You wanted Pl3xMap support. I could not find Pl3xMap.");
+            this.pl3xMap = null;
+            this.enabled = false;
+        }
+    }
+
+    @Override
+    public void onVanish(@NonNull Player player) {
+        if(this.enabled && this.pl3xMap != null) {
```
Comment (mbax) [inline src/main/java/org/kitteh/vanish/hooks/plugins/Pl3xMapHook.java:35]: Space!

File: src/main/java/org/kitteh/vanish/hooks/plugins/Pl3xMapHook.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            this.pl3xMap.playerManager().hidden(player.getUniqueId(), true);
+        }
+    }
+
+    @Override
+    public void onUnvanish(@NonNull Player player) {
+        if(this.enabled && this.pl3xMap != null) {
```
Comment (mbax) [inline src/main/java/org/kitteh/vanish/hooks/plugins/Pl3xMapHook.java:42]: image

**Review by mbax — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (AAigars) [reply pom.xml]: Not entirely sure, why I did bump it but it has been reverted :)

**Comment by mbax** [comment]
#704

**Pull request closed without merging**